homeassistant/components/starlink/coordinator.py

```diff
@@ -0,0 +1,61 @@
+"""Contains the shared Coordinator for Starlink systems."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from starlink_grpc import ChannelContext, GrpcError, get_status
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .dish_status import DishyStatus
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class StarlinkUpdateCoordinator(DataUpdateCoordinator[DishyStatus]):
+    """Coordinates updates between all Starlink sensors defined in this file."""
+
+    def __init__(self, hass: HomeAssistant, name: str, url: str) -> None:
+        """Initialize an UpdateCoordinator for a group of sensors."""
+        self.channel_context = ChannelContext(target=url)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=1),
```

Comment:
Do you actually need to poll every second? Looking at exposed sensors I don't think it makes sense to update any of them more than once per minute. Maybe even less often

Reply:
1 second matches the Starlink app, any longer and we risk missing events for more critical sensors such as obstruction status (which aren't in this PR because that's a different platform). If you have a better solution, I'm all ears 🙂

Reply:
Closing this as no better solutions have been provided

Reply:
Regardless, it should be user configurable, right?